server/expression/explicit_cast.go

```diff
@@ -84,9 +84,12 @@ func (c *ExplicitCast) Eval(ctx *sql.Context, row sql.Row) (any, error) {
 	if val == nil {
 		return nil, nil
 	}
+	if fromType.BaseID() == pgtypes.DoltgresTypeBaseID_Unknown {
+		fromType = pgtypes.Text
+	}
```

Comment:
This should come after the `castFunction == nil` check, similar to how `assignment_cast.go` was changed.